src/test/java/integration/IntegrationTest.java

```diff
@@ -3,563 +3,1428 @@
+        testFileChange(ws1, file1, new WSManager[]{}); // Test User1 changing File1; Should have 6 changes on File1, 3 change on File2
+        testProjectSubscribe(ws2, proj1);
+
+        testFileDeleteInvalid();
+        testFileDelete(ws2, file1, new WSManager[]{ws1});
+        testFileDelete(ws2, file2, new WSManager[]{});
+        testProjectGetFiles(); // should have 0 files each
+
+        testProjectLookup(ws1, user1ID, proj1);
+        testProjectLookup(ws2, user2ID, proj2);
+
+        testProjectDelete(ws1, proj1, new WSManager[]{ws2});
+        testProjectDelete(ws2, proj2, new WSManager[]{});
+
+        testUserProjects();
+
+        testUserDelete(ws1, user1ID, user1Pass);
+        testUserDelete(ws2, user2ID, user2Pass);
+
+        testUserLogin(ws3, user3ID, user3Pass);
+        testUserDelete(ws3, user3ID, user3Pass);
+
+        Thread.sleep(1000);
+    }
+
+    private void testUserRegister() throws InterruptedException, ConnectException {
+        Semaphore waiter = new Semaphore(0);
+
+        req = new UserRegisterRequest(user1ID, user1FirstName, user1LastName, user1Email, user1Pass).getRequest(response -> {
+            // If registration fails, probably is already there.
+            Assert.assertNotEquals(String.format("user %s already registered, rerun test", user1ID), 404, response.getStatus());
+            Assert.assertEquals(String.format("Failed to register user: %s", user1ID), 200, response.getStatus());
+
+            waiter.release();
+        }, errHandler);
+
+        ws1.sendRequest(req);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("testUserRegister timed out");
+        }
+
+        req = new UserRegisterRequest(user2ID, user2FirstName, user2LastName, user2Email, user2Pass).getRequest(response -> {
+            // If registration fails, probably is already there.
+            Assert.assertNotEquals(String.format("user %s already registered, rerun test", user2ID), 404, response.getStatus());
+            Assert.assertEquals(String.format("Failed to register user: %s", user2ID), 200, response.getStatus());
+
+            waiter.release();
+        }, errHandler);
+
+        ws2.sendRequest(req);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("testUserRegister timed out");
+        }
+
+        req = new UserRegisterRequest(user3ID, user3FirstName, user3LastName, user3Email, user3Pass).getRequest(response -> {
+            // If registration fails, probably is already there.
+            Assert.assertNotEquals(String.format("user %s already registered, rerun test", user3ID), 404, response.getStatus());
+            Assert.assertEquals(String.format("Failed to register user: %s", user3ID), 200, response.getStatus());
+
+            waiter.release();
+        }, errHandler);
+
+        ws2.sendRequest(req);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("testUserRegister timed out");
+        }
```

Comment:
This whole method seems like a lot of duplicate code. A single testUserRegister() that registers all three should probably still exist, but it should call a helper method three times that does a single registration.